kostyor/db/api.py

```diff
@@ -54,6 +52,26 @@ def create_discovery_method(method):
 
 def create_cluster_upgrade(cluster_id, to_version):
     cluster = db_session.query(models.Cluster).get(cluster_id)
+
+    if cluster is None:
+        raise exceptions.ClusterNotFound(
+            'Cluster "%s" not found.' % cluster_id)
+
+    if cluster.version == constants.UNKNOWN:
+        raise exceptions.ClusterVersionIsUnknown('Cluster version is unknown')
+
+    if cluster.status == constants.UPGRADE_IN_PROGRESS:
+        raise exceptions.UpgradeIsInProgress(
+            'Cluster %s already has an upgrade in progress.' % cluster_id)
```

Comment:
These checks are duplicates of the logic in

https://github.com/sc68cal/Kostyor/pull/39/files#diff-8a04877a7bc46c5eff7b889c49eab0a3R171

Which is executed prior to the call into this function. So, either remove this new piece or remove the checks in the REST api file.